source/grid/tria_description.cc

```diff
@@ -741,42 +741,47 @@ namespace TriangulationDescription
       const TriangulationDescription::Settings    settings,
       const unsigned int                          my_rank_in)
     {
-      if (const auto tria_pdt = dynamic_cast<
-            const parallel::distributed::Triangulation<dim, spacedim> *>(&tria))
-        Assert(comm == tria_pdt->get_communicator(),
-               ExcMessage("MPI communicators do not match."));
-
-      // First, figure out for what rank we are supposed to build the
-      // TriangulationDescription::Description object
-      unsigned int my_rank = my_rank_in;
-      Assert(my_rank == numbers::invalid_unsigned_int ||
-               my_rank < dealii::Utilities::MPI::n_mpi_processes(comm),
-             ExcMessage("Rank has to be smaller than available processes."));
-
-      if (auto tria_pdt = dynamic_cast<
-            const parallel::distributed::Triangulation<dim, spacedim> *>(&tria))
+      if (const auto ptria =
+            dynamic_cast<const parallel::TriangulationBase<dim, spacedim> *>(
+              &tria))
         {
-          Assert(my_rank == numbers::invalid_unsigned_int ||
-                   my_rank == dealii::Utilities::MPI::this_mpi_process(comm),
+          Assert(comm == ptria->get_communicator(),
+                 ExcMessage("MPI communicators do not match."));
```

Comment:
Do we always check for communicators to be equal rather than using `MPI_Comm_compare()`? This was pre-existing, but we might want to enforce one or the other in the library in another patch.

Reply:
Good question. I don't know. We played with letting each object duplicate the communicator many years ago, to make sure that we don't run into race conditions: Every object has its own communicator to communicate over. That turned out not to be practical because we ended up creating millions and billions of communicators, and then destroying them again. But at least conceptually I think that it is an option we would like to support.

If you feel strongly about it, feel free to open a github issue. In any case, I think that's for a separate patch as you mention.